core/src/main/java/com/linecorp/armeria/common/HttpRequestBuilder.java

```diff
@@ -0,0 +1,417 @@
+        this.publisher = null;
+        return this;
+    }
+
+    /**
+     * Sets the content for this request. The content of the request is formatted by
+     * {@link String#format(Locale, String, Object...)} with {@linkplain Locale#ENGLISH English locale}.
+     */
+    @FormatMethod
+    public HttpRequestBuilder content(MediaType mediaType, @FormatString String format, Object... content) {
+        requireNonNull(mediaType, "mediaType");
+        requireNonNull(format, "format");
+        requestHeadersBuilder.contentType(mediaType);
+        this.content = HttpData.of(mediaType.charset(StandardCharsets.UTF_8), format, content);
+        this.publisher = null;
+        return this;
+    }
+
+    /**
+     * Sets the content for this request.
+     */
+    public HttpRequestBuilder content(MediaType mediaType, byte[] content) {
+        requireNonNull(mediaType, "mediaType");
+        requireNonNull(content, "content");
+        requestHeadersBuilder.contentType(mediaType);
+        this.content = HttpData.wrap(content);
+        this.publisher = null;
+        return this;
+    }
+
+    /**
+     * Sets the content for this request.
+     */
+    public HttpRequestBuilder content(MediaType mediaType, HttpData content) {
+        requireNonNull(mediaType, "mediaType");
+        requireNonNull(content, "content");
+        requestHeadersBuilder.contentType(mediaType);
+        this.content = content;
+        this.publisher = null;
+        return this;
+    }
+
+    /**
+     * Sets the content for this request.
+     */
+    public HttpRequestBuilder content(MediaType mediaType, Publisher<HttpData> publisher) {
+        requireNonNull(mediaType, "mediaType");
+        requireNonNull(publisher, "publisher");
+        requestHeadersBuilder.contentType(mediaType);
+        this.publisher = publisher;
+        this.content = null;
+        return this;
+    }
+
+    /**
+     * Sets a header for this request. For example:
+     * <pre>{@code
+     * HttpRequest.builder()
+     *            .get("/")
+     *            .header("authorization", "foo")
+     *            .build();
+     * }</pre>
+     */
+    public HttpRequestBuilder header(CharSequence name, Object value) {
+        requireNonNull(name, "name");
+        requireNonNull(value, "value");
+        requestHeadersBuilder.setObject(name, value);
+        return this;
+    }
+
+    /**
+     * Sets multiple headers for this request. For example:
+     * <pre>{@code
+     * HttpRequest.builder()
+     *            .get("/")
+     *            .headers(HttpHeaders.of("authorization", "foo", "bar", "baz"))
+     *            .build();
+     * }</pre>
+     *
+     * @see HttpHeaders
+     */
+    public HttpRequestBuilder headers(HttpHeaders httpHeaders) {
+        requireNonNull(httpHeaders, "httpHeaders");
+        for (AsciiString name : httpHeaders.names()) {
+            requestHeadersBuilder.set(name, requireNonNull(httpHeaders.get(name), "header"));
```

Comment:
Use `httpHeaders.forEach(...)` for:
- We can remove additional hash lookup costs.
- The value is not null always.